code/modules/spells/targeted/genetic.dm

```diff
@@ -105,15 +105,12 @@ code\game\\dna\genes\goon_powers.dm
 	var/mob/living/carbon/human/H = user
 	var/list/trays = list()
 	for(var/obj/machinery/portable_atmospherics/hydroponics/tray in range(1))
-		if(tray.weedlevel > 0)
+		if(tray.get_weedlevel() > 0)
 			trays += tray
 
-	var/obj/machinery/portable_atmospherics/hydroponics/target = input(H,"Select a tray:") as null|anything in trays
+	var/obj/machinery/portable_atmospherics/hydroponics/target = pick(trays)
```

Comment:
technically a function change but honestly this version will probably feel more comfortable anyway since most people just mash the power button as is

Reply:
All trays are called hydroponics tray, so I don't think it is useful... Unless someone labels them differently